### PR TITLE
RD-4296 - mimic AWS import strategy of the original handler

### PR DESCRIPTION
**Pull request by saartochner**

### CODE REVIEW

File: src/lumigo_tracer/auto_instrument_handler.py [hunk 1]
```diff
@@ -10,13 +10,13 @@ def parse_handler():
...
     except KeyError:
-        raise ValueError(
+        raise Exception(
             "Could not find the original handler. Please contact Lumigo for more information."
-        )
-    except ValueError:
-        raise RuntimeError(
-            f"Invalid handler format: Bad handler '{os.environ[ORIGINAL_HANDLER_KEY]}': not enough values to unpack (expected 2, got 1)"
-        )
+        ) from None
```
Comment (nirLumigo) [inline src/lumigo_tracer/auto_instrument_handler.py:15]: Why are you adding `from None`? The user will lose the context of the exception:
image

Same regarding the other `from None` is this file

File: src/lumigo_tracer/auto_instrument_handler.py [hunk 2]
```diff
@@ -26,11 +26,21 @@ def _handler(*args, **kwargs):
...
     try:
         handler_module, unit_name = parse_handler()
-        original_handler = getattr(importlib.import_module(handler_module), unit_name)
-    except (ImportError, AttributeError):
+        original_module = importlib.import_module(handler_module)
+    except ImportError as e:
         raise ImportError(
-            f"Unable to import module '{handler_module}': No module named '{handler_module}'"
-        )
+            f"Runtime.ImportModuleError: Unable to import module '{handler_module}': {str(e)}"
```
Comment (nirLumigo) [inline src/lumigo_tracer/auto_instrument_handler.py:32]: Here you even add of original exception in the text. Don't you prefer to do `from e` instead?

File: src/test/unit/test_auto_instrument_handler.py [hunk 3]
```diff
@@ -32,33 +32,67 @@ def test_hierarchy_happy_flow(monkeypatch):
 def test_import_error(monkeypatch):
     monkeypatch.setenv(ORIGINAL_HANDLER_KEY, "blabla.not.exists")
 
-    with pytest.raises(ImportError):
```
Comment (nirLumigo) [inline src/test/unit/test_auto_instrument_handler.py:35]: Why not do it with `pytest.raises`?
Same issue regarding the other tests in this file as well.

**Review by nirLumigo — CHANGES_REQUESTED** [review]
Nice :)

File: src/lumigo_tracer/auto_instrument_handler.py (hunk 1 above)
Comment (saartochner) [reply src/lumigo_tracer/auto_instrument_handler.py:15]: Yeah, That's what AWS does, and I want to have the same behavior...

File: src/lumigo_tracer/auto_instrument_handler.py (hunk 2 above)
Comment (saartochner) [reply src/lumigo_tracer/auto_instrument_handler.py:32]: I wanted to copy AWS's bootstrap default behavior: 
image

File: src/test/unit/test_auto_instrument_handler.py (hunk 3 above)
Comment (saartochner) [reply src/test/unit/test_auto_instrument_handler.py:35]: That's a great question!
I did use pytest.raises in the first place, but then I noticed that it removes the context of the message, and I want to validate that there was no context (to follow AWS behavior!) :D

File: src/lumigo_tracer/auto_instrument_handler.py (hunk 1 above)
Comment (saartochner) [reply src/lumigo_tracer/auto_instrument_handler.py:15]: Yep, I tried to copied AWS behavior:
image

File: src/test/unit/test_auto_instrument_handler.py (hunk 3 above)
Comment (nirLumigo) [reply src/test/unit/test_auto_instrument_handler.py:35]: Add a comment explaining why don't we use it

**Review by nirLumigo — APPROVED** [review]

**Pull request merged**